src/button-group/menu-dropdown-item.tsx

```diff
@@ -0,0 +1,77 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React from 'react';
+import clsx from 'clsx';
+
+import InternalButton from '../button/internal';
+import { ButtonDropdownProps } from '../button-dropdown/interfaces';
+import ButtonDropdown from '../button-dropdown/internal';
+import Tooltip from '../internal/components/tooltip';
+import { CancelableEventHandler, fireCancelableEvent } from '../internal/events';
+import { ButtonGroupProps } from './interfaces';
+
+import styles from './styles.css.js';
+import testUtilStyles from './test-classes/styles.css.js';
+
+const MenuDropdownItem = React.forwardRef(
+  (
+    {
+      item,
+      showTooltip,
+      onItemClick,
+      expandToViewport,
+    }: {
+      item: ButtonGroupProps.MenuDropdown;
+      showTooltip: boolean;
+      onItemClick?: CancelableEventHandler<ButtonGroupProps.ItemClickDetails>;
+      expandToViewport?: boolean;
+    },
+    ref: React.Ref<ButtonDropdownProps.Ref>
+  ) => {
+    const containerRef = React.useRef<HTMLDivElement>(null);
+    const onClickHandler = (event: CustomEvent<ButtonDropdownProps.ItemClickDetails>) => {
+      fireCancelableEvent(onItemClick, { id: event.detail.id }, event);
+    };
+
+    return (
+      <ButtonDropdown
+        ref={ref}
+        variant="icon"
+        loading={item.loading}
+        loadingText={item.loadingText}
+        disabled={item.disabled}
+        items={item.items}
+        onItemClick={onClickHandler}
+        expandToViewport={expandToViewport}
+        ariaLabel={item.text}
+        className={testUtilStyles['button-group-item']}
+        data-testid={item.id}
+        customTriggerBuilder={({ onClick, isOpen, triggerRef, ariaLabel, ariaExpanded, testUtilsClass }) => (
+          <div ref={containerRef}>
+            {!isOpen && showTooltip && !item.disabled && !item.loading && (
+              <Tooltip
+                trackRef={containerRef}
+                trackKey={item.id}
+                value={item.text}
+                className={clsx(styles.tooltip, testUtilStyles['button-group-tooltip'])}
+              />
+            )}
+            <InternalButton
+              ref={triggerRef}
+              variant="icon"
+              ariaLabel={ariaLabel}
+              ariaExpanded={ariaExpanded}
+              className={clsx(styles.item, testUtilsClass)}
+              data-testid={item.id}
```

Comment:
We are setting the same `data-test-id` twice (in this line and above, in line 48) - can we avoid setting the same ID twice?

Reply:
Do you see a risk with that?

These test-IDs are always used in combination with class names. We need the one on the button-dropdown so that the test-utils elector can point to the component's root so that we can wrap it with ButtonDropdownWrapper.

The second one is used in the internal.tsx to find the focus targets that are the actual focusable elements so we need it to point to the trigger. We can technically find the button dropdown and use a nested selector to find a trigger but that would increase those queries complexity by much as we'd need to assert item types to figure out the selector to use.

Reply:
@pan-kot I think we can keep only one. Am I right?

Reply:
Addressed here: https://github.com/cloudscape-design/components/pull/2469/commits/4a40ee45296b5b217aeb03a14abe0156f51feaab